x-pack/plugins/fields_metadata/README.md

```diff
@@ -0,0 +1,168 @@
+# Fields Metadata Plugin
+
+The `@kbn/fields-metadata-plugin` is designed to provide a centralized and asynchronous way to consume field metadata across Kibana. This plugin addresses the need for on-demand retrieval of field metadata from static ECS definitions and integration manifests, with the flexibility to extend to additional resolution sources in the future.
+
+## Components and Mechanisms
+
+### FieldsMetadataService (Server-side)
+
+The `FieldsMetadataService` is instantiated during the plugin setup/start lifecycle on the server side. It exposes a client that can be used to consume field metadata and provides tools for registering external dependencies.
+
+#### Start Contract
+
+The start contract exposes a `FieldsMetadataClient` instance, which offers the following methods:
+- `getByName(name: string, params? {integration: string, dataset?: string})`: Retrieves a single `FieldMetadata` instance by name.
+
+```ts
+const timestampField = await client.getByName('@timestamp')
+/*
+{
+  dashed_name: 'timestamp',
+  type: 'date',
+  ...
+}
+*/
+```
+
+- `find(params?: {fieldNames: string[], integration: string, dataset?: string})`: Retrieves a record of matching `FieldMetadata` instances based on the query parameters.
+
+**Parameters**
+| Name | Type | Example | Optional |
+|---|---|---|---|
+| fieldNames | <EcsFieldName \| string>[] | ['@timestamp', 'onepassword.client.platform_version'] | ✅ |
+| integration | string | 1password | ✅ |
+| dataset | string | 1password.item_usages | ✅ |
+
+```ts
+const fields = await client.find({
+  fieldNames: ['@timestamp', 'onepassword.client.platform_version'], 
+  integration: '1password'
+})
+/*
+{
+  '@timestamp': {
+    dashed_name: 'timestamp',
+    type: 'date',
+    ...
+  },
+  'onepassword.client.platform_version': {
+    name: 'platform_version',
+    type: 'keyword',
+    ...
+  },
+}
+*/
+```
+
+> N.B. Passing the `dataset` name parameter to `.find` helps narrowing the scope of the integration assets that need to be fetched, increasing the performance of the request. Using only the `integration` parameter should achieve the same result, but is recommended always passing the `dataset` as well if known or unless the required fields come from different datasets of the same integration.
```

Comment:
It might be a good idea to make this a required value if you want API consumers to consider the best option, with `all` or `*` to indicate any dataset.

Reply:
I applied your suggestion on the `dataset` params, it makes sense to enforce the best option and still leave the door open to cases where the dataset is unknown 👌 I opted for suggesting the `*` value to access any integration dataset fields on the target integration and constrained its presence when `integration` is passed.

[refactor(fields-metadata): make dataset parameter required if integra…](https://github.com/elastic/kibana/pull/183806/commits/9b132213d1f9cf79e25275feb14e4a1c0106d908)

Regarding `fieldNames` and `attributes`, they are mostly used to filter the returned data, I still believe they should be optional with the default behaviour of returning a default set.